test/move_effect_toxic_spikes.c

```diff
@@ -0,0 +1,120 @@
+        EXPECT(opponent[0].status1 & STATUS1_TOXIC_POISON);
+    }
+}
+
+SINGLE_BATTLE_TEST("Abilities prevent toxic spike effect.")
+{
+    u16 ability;
+    PARAMETRIZE { ability = ABILITY_PASTEL_VEIL; }
+    PARAMETRIZE { ability = ABILITY_IMMUNITY; }
+    PARAMETRIZE { ability = ABILITY_LEVITATE; }
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        switch (ability) {
+        case ABILITY_PASTEL_VEIL:
+            OPPONENT(SPECIES_PONYTA_GALARIAN) { Ability(ability); };
+            break;
+        case ABILITY_IMMUNITY:
+            OPPONENT(SPECIES_SNORLAX) {Ability(ability); };
+            break;
+        case ABILITY_LEVITATE:
+            OPPONENT(SPECIES_UNOWN) {Ability(ability); };
+            break;
+        }
+    } WHEN {
+        TURN { MOVE(player, MOVE_TOXIC_SPIKES); }
+        TURN { SWITCH(opponent, 1); }
+    } THEN {
+        EXPECT(gSideTimers[B_SIDE_OPPONENT].toxicSpikesAmount == 1);
+        EXPECT(!(opponent->status1 & STATUS1_POISON));
+    }
+}
+
+// FIXME: Test Gravity, Ingrain (From Baton Bass)
+SINGLE_BATTLE_TEST("Balloon prevents toxic spike effect")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) {Item(ITEM_AIR_BALLOON); };
+    } WHEN {
+        TURN { MOVE(player, MOVE_TOXIC_SPIKES); }
+        TURN { SWITCH(opponent, 1); }
+    } THEN {
+        EXPECT(gSideTimers[B_SIDE_OPPONENT].toxicSpikesAmount == 1);
+        EXPECT(!(opponent[0].status1 & STATUS1_POISON));
+    }
+}
+
+// FIXME: Test Gravity, Ingrain (From Baton Bass)
+SINGLE_BATTLE_TEST("Types prevent toxic spike effect")
+{
+    u32 type;
+    u32 item;
+    PARAMETRIZE { type = TYPE_STEEL; item = ITEM_NONE; }
+    PARAMETRIZE { type = TYPE_FLYING; item = ITEM_NONE; }
+    PARAMETRIZE { type = TYPE_FLYING; item = ITEM_IRON_BALL;  }
+    PARAMETRIZE { type = TYPE_POISON; item = ITEM_NONE; }
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        switch (type) {
+        case TYPE_STEEL:
+            OPPONENT(SPECIES_STEELIX) { Item(item); };
+            break;
+        case TYPE_FLYING:
+            OPPONENT(SPECIES_PIDGEOT) { Item(item); };
+            break;
+        case TYPE_POISON:
+            OPPONENT(SPECIES_EKANS) { Item(item); };
+            break;
+        }
+    } WHEN {
+        TURN { MOVE(player, MOVE_TOXIC_SPIKES); }
+        if (type == TYPE_POISON) {
+            TURN { MOVE(opponent, MOVE_MAGNET_RISE); }
+        }
+        TURN { SWITCH(opponent, 1); }
+    } THEN {
+        if (type != TYPE_POISON)
+            EXPECT(gSideTimers[B_SIDE_OPPONENT].toxicSpikesAmount == 1);
+        else
+            EXPECT(gSideTimers[B_SIDE_OPPONENT].toxicSpikesAmount == 0);
+        
+        if (item != ITEM_IRON_BALL)
+            EXPECT(!(opponent->status1 & STATUS1_POISON));
+        else
+            EXPECT(opponent->status1 & STATUS1_POISON);
+    }
+}
```

Comment:
I'd be tempted to reorganize these into:
- Things that make the battler non-grounded prevent toxic spikes effect (Flying, Levitate, Magnet Rise). Canceled out by Iron Ball and Gravity (maybe that should be a separate test—there's quite a few things that care about being grounded).
- Things that make the battler immune to poison prevent toxic spikes effect (Steel, Immunity, Pastel Veil).
- Grounded Poison-types clear toxic spikes.